spec/data-model/README.md

```diff
@@ -110,38 +110,48 @@ interface CatchallKey {
 ## Patterns
 
 Each `Pattern` represents a linear sequence, without selectors.
-Each element of the `body` array MUST either be a non-empty string or an `Expression` object.
-String values represent literal _text_,
-while `Expression` wraps each of the potential _expression_ shapes.
+Each element of the `body` array MUST be a non-empty string, an `Expression` or a `Markup` object.
+String values represent literal _text_.
+`Expression` wraps each of the potential _expression_ shapes.
+`Markup` wraps each of the potential _markup_ shapes.
 The `body` strings are the "cooked" _text_ values, i.e. escape sequences are processed.
 
-Implementations MUST NOT rely on the set of `Expression` interfaces being exhaustive,
-as future versions of this specification MAY define additional expressions.
+Implementations MUST NOT rely on the set of `Expression` and 
+`Markup` interfaces defined in this document being exhaustive.
+Future versions of this specification might define additional
+types of expression or markup.
```

Comment:
Minor wording. "Types of" is unncessary and also misleading, since we might use the same "type" of expression for a new function (for example)

```suggestion
expressions or markup.
```